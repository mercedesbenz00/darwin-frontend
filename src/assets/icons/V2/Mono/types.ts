export enum IconVariant {
  DEFAULT = 'default',
  DISABLED = 'disabled',
  SELECTED = 'selected',
  NEGATIVE = 'negative',
}
