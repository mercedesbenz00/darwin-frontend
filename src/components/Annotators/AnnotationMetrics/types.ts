import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
