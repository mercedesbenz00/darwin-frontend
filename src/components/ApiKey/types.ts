export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
