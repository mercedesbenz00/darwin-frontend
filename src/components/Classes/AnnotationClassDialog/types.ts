import { AnnotationClassPayload, AnnotationClassMetadata, AnnotationTypeName } from '@/store/types'

type PendingFields =
  | 'crop_key'
  | 'crop_url'
  | 'image_height'
  | 'image_width'
  | 'key'
  | 'original_image_url'
  | 'scale'
  | 'x'
  | 'y'

/**
 * Holds a class image that isn't necessarily persisted to the backend yet.
 * That means it might be missing any of the url fields
 *
 * It can be saved to backend once at least the `key` field is set, as that one
 * can be used to infer/compute all the other fields.
 *
 * - `crop_key` - repleace "original" with "crop" in `key` field
 * - `crop_url` - signed s3 url (by backend), for `crop_key`
 * - `original_image_url` - signed s3 url (by backend) for `key`
 */
export type PendingClassImage =
  Omit<AnnotationClassPayload['images'][0], PendingFields>
  & Partial<Pick<AnnotationClassPayload['images'][0], PendingFields>>

// eslint-disable-next-line camelcase
export type PendingClassImageWithUrlSet = PendingClassImage & { original_image_url: string }

export const isClassImage = (
  image: PendingClassImage
): image is AnnotationClassPayload['images'][0] =>
  !!image.key && !!image.crop_key

export type AnnotationClassData = {
  datasets: AnnotationClassPayload['datasets']
  description: string | null
  hotkey: null | string
  id: number | null
  images: PendingClassImage[]
  metadata: AnnotationClassMetadata
  name: string | null
}

export type DefaultAnnotationClassData = Partial<AnnotationClassData> & {
  selectedMainAnnotationType?: AnnotationTypeName
}

export type AnnotationClassValidationErrors = {
  annotationTypes?: string
  name?: string
  skeleton?: string
}

export const VIEWPORT_WIDTH: number = 150
export const VIEWPORT_HEIGHT: number = 150

export type CropInfo = {
  orientation: number
  points: [string, string, string, string]
  zoom: number
}

/**
 * VueCroppie is really archaic and requires us to get crop info via $ref,
 * so this type defines that ref
 */
export type Croppie = {
  /** Binds options to the underlying croppie library */
  bind: (options: Pick<CropInfo, 'points' | 'zoom'> & { url: string}) => void

  /** Retrieves current crop info */
  get: () => CropInfo
  result: (options?: Record<string, string>, callback?: Function) => void

  /** Sets scale/zoom level of cropper */
  setZoom: (value: number) => void
}
