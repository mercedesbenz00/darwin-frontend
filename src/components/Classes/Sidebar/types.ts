import { AnnotationTypeName } from '@/store/types'

export type AnnotationTypeFilterItemType = {
  id: string | number
  label: string
  icon?: string
  name: AnnotationTypeName
  color: string
  count: number | null
  [key: string]: any
}
