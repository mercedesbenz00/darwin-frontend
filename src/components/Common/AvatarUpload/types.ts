export type AvatarUploadData = {
  hash: string
  file: File
  type: string | null
}
