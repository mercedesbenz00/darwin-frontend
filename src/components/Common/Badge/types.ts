import { RGBA } from '@/utils'

export type ParsedColorType = {
  bg: string,
  text: string
}

export type BadgeType = {
  id?: string,
  label: string,
  color?: RGBA
  size?: 'small' | 'medium' | 'large'
  highContrast?: boolean
}

export enum BadgeSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}

export enum BadgeTag {
  A = 'a',
  DIV = 'div',
  ROUTER_LINK = 'router-link',
  BUTTON = 'button'
}

export interface BadgesProps {
  items: BadgeType[],
  selected?: boolean,
  wrap?: boolean
}

export interface BadgeProps {
  label: string,
  color?: RGBA,
  size?: string,
  tag?: string,
  tooltip?: string,
  highContrast?: boolean,
  overrideLabelColor?: string,
  deletable?: boolean,
  noTooltip?: boolean,
  alpha?: number
}
