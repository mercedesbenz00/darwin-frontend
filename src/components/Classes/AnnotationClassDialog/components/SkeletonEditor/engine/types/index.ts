export * from './SkeletonEdgeType'
export * from './SkeletonNodeType'
