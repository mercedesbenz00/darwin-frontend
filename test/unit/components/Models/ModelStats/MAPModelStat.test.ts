import { shallowMount } from '@vue/test-utils'

import { buildTrainedModelPayload } from 'test/unit/factories'

import MAPModelStat from '@/components/Models/ModelStats/MAPModelStat.vue'
import { TrainedModelPayload } from '@/store/types'

let propsData: {
  iouValue: string | null
  trainedModel: TrainedModelPayload | null
}

const itMatchesSnapshot = () => it('matches snapshot', () => {
  const wrapper = shallowMount(MAPModelStat, { propsData })
  expect(wrapper).toMatchSnapshot()
})

describe('when trainedModel is null', () => {
  beforeEach(() => {
    propsData = { iouValue: null, trainedModel: null }
  })

  itMatchesSnapshot()
})

describe('when training results are null', () => {
  beforeEach(() => {
    propsData = {
      iouValue: null,
      trainedModel: buildTrainedModelPayload({
        training_result: null
      })
    }
  })

  itMatchesSnapshot()
})

describe('when training results segm field in undefined', () => {
  beforeEach(() => {
    propsData = {
      iouValue: null,
      trainedModel: buildTrainedModelPayload({
        training_result: {}
      })
    }
  })

  itMatchesSnapshot()
})

describe('when iouValue is null, use AP', () => {
  beforeEach(() => {
    propsData = {
      iouValue: null,
      trainedModel: buildTrainedModelPayload({
        training_result: {
          segm: {
            AP: 50,
            AP50: 75,
            AP75: 60
          }
        }
      })
    }
  })

  itMatchesSnapshot()
})

describe('when iouValue is 50%, use AP50', () => {
  beforeEach(() => {
    propsData = {
      iouValue: '50%',
      trainedModel: buildTrainedModelPayload({
        training_result: {
          segm: {
            AP: 50,
            AP50: 75,
            AP75: 60
          }
        }
      })
    }
  })

  itMatchesSnapshot()
})

describe('when iouValue is 75%, use AP75', () => {
  beforeEach(() => {
    propsData = {
      iouValue: '75%',
      trainedModel: buildTrainedModelPayload({
        training_result: {
          segm: {
            AP: 50,
            AP50: 75,
            AP75: 60
          }
        }
      })
    }
  })

  itMatchesSnapshot()
})

describe('when iouValue encodes any other value, use AP', () => {
  beforeEach(() => {
    propsData = {
      iouValue: 'Average',
      trainedModel: buildTrainedModelPayload({
        training_result: {
          segm: {
            AP: 50,
            AP50: 75,
            AP75: 60
          }
        }
      })
    }
  })

  itMatchesSnapshot()
})
