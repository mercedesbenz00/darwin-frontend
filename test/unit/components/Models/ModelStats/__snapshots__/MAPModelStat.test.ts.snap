// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when iouValue encodes any other value, use AP matches snapshot 1`] = `
<model-stat-stub name="mAP" score="50%">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when iouValue is 50%, use AP50 matches snapshot 1`] = `
<model-stat-stub name="mAP" score="75%">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when iouValue is 75%, use AP75 matches snapshot 1`] = `
<model-stat-stub name="mAP" score="60%">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when iouValue is null, use AP matches snapshot 1`] = `
<model-stat-stub name="mAP" score="50%">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when trainedModel is null matches snapshot 1`] = `
<model-stat-stub name="mAP" score="N/A">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when training results are null matches snapshot 1`] = `
<model-stat-stub name="mAP" score="N/A">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;

exports[`when training results segm field in undefined matches snapshot 1`] = `
<model-stat-stub name="mAP" score="N/A">
  <map-hero-stub></map-hero-stub>
  <p><strong class="bold">Mean Average Precision</strong>
    is measured by taking the mean of all average precisions (the area under a Precision vs
    Recall curve) across all IoU thresholds and for all classes. This metric provides an
    overall model performance, irrespective of any manually set threshold. MaP is a harsh
    measurement, so don’t be scared if its number is low.
  </p>
</model-stat-stub>
`;
